### PR TITLE
P6

### PR DESCRIPTION
**Pull request by PeteChon1**
Hi guys,

Please review the solution to problem 6 in JS1 when you get the time.  Thanks!

### CODE REVIEW

File: curriculum/js1/6.js [hunk 1]
```diff
@@ -5,8 +5,15 @@
  * @returns {number}
  */
 
-const solution = (a) => {
-  return 0;
+const solution = (a, possibleDivisor = a -1, sum = 0) => {
+  if (possibleDivisor === 1) {
+    return sum;
+  }
```
Comment (danielthalhuber) [inline curriculum/js1/6.js]: This looks good, my only recommendation is to put it all on one line like this to clean up the presentation/readability (completely optional):
```javascript
if (possibleDivisor === 1) return sum;
```

File: curriculum/js1/6.js [hunk 2]
```diff
@@ -5,8 +5,15 @@
...
+  if (a % possibleDivisor === 0) {
+    sum = sum + possibleDivisor;
+  }
```
Comment (danielthalhuber) [inline curriculum/js1/6.js]: This is good. If you're looking for ways to improve/shorten the code, consider:
* condensing everything onto one line
* use the [addition assignment operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Assignment_Operators#Addition_assignment)
```javascript
if (a % possibleDivisor === 0) sum += possibleDivisor;
```

File: curriculum/js1/6.js [hunk 3]
```diff
@@ -5,8 +5,15 @@
...
+  possibleDivisor = possibleDivisor - 1;
+  return solution(a, possibleDivisor, sum);
```
Comment (danielthalhuber) [inline curriculum/js1/6.js]: This works. I can only offer options for shortening the code. If you want, you could do this arithmetic inside of the function call:
```javascript
return solution(a, possibleDivisor - 1, sum);
```

File: curriculum/js1/6.js [hunk 4]
```diff
@@ -5,8 +5,15 @@
  * @returns {number}
  */
 
-const solution = (a) => {
-  return 0;
+const solution = (a, possibleDivisor = a -1, sum = 0) => {
```
Comment (danielthalhuber) [inline curriculum/js1/6.js]: This is super-minor, but I would add another space here:
```javascript
possibleDivisor = a - 1
```

**Review by danielthalhuber — APPROVED** [review]

**Review by songz — APPROVED** [review]

**Pull request merged**